src/SnoopCompile.jl

```diff
@@ -25,58 +24,36 @@ function snoop(filename, commands)
     # addprocs will run the unmodified version of julia, so we
     # launch it as a command.
     code_object = """
-        while !eof(STDIN)
-            eval(Main, deserialize(STDIN))
+            using Serialization
+            while !eof(stdin)
+                Core.eval(Main, deserialize(stdin))
+            end
+            """
+    process = open(`$(Base.julia_cmd()) --eval $code_object`, stdout, write=true)
```

Comment:
Could we maybe allow adding some optional flags like `--project` here?  Maybe just allow passing an iterable of flags to `@snoop`, which will be interpolated into the command.